Gh-358: cover remote coding environments in dev docs (#375)
gchq/gaffer-doc: docs/dev/remote-coding-environments.md

# Remote Coding Environments For Gaffer

Gaffer is now configured for remote coding environemnts such as GitHub
Codespaces and Gitpod. This addition allows for an easier and faster way to

> contribute to Gaffer with no manual setup or need to download dependices to a

Reviewer: ```suggestion
contribute to Gaffer with no manual setup or need to download dependencies to a
```

local machine.

## GitHub Codespaces

> To use GitHub Codespaces simply open the gaffer repsoitory and click the "Code"
> button drop down. You'll see an option labeled "Open with codespaces" Click on
> it.

Reviewer: ```suggestion
To use GitHub Codespaces simply open the Gaffer repository, click the "Code" button
drop down, and then the option labeled "Open with Codespaces".
```
However, this is not what I see. I see a tab called codespaces and a green "Create codespace on develop" button.

> This will launch Codespaces with all the correct configuration needed to
> contribute to the Gaffer repository.
> 
> For more information on GitHub Codespaces click
> [here](https://github.com/features/codespaces).

Reviewer: ```suggestion
This will launch a Codespaces environment with all the configuration needed
to contribute to Gaffer.
See the [GitHub documentation for more information on Codespaces]
(https://github.com/features/codespaces).
```

## Gitpod
To use Gitpod you can simply prefix any GitHub URL with **`gitpod.io/#`** and
follow the steps from there. You can also install the
[extension](https://www.gitpod.io/docs/configure/user-settings/browser-extension)
on your web browser. This adds a button to GitHub that does the prefixing for
you.

Once on Gitpod, your environment will be set up with all the correct

> configuration needed to contribute to the Gaffer repository.

Reviewer: We should also include a bit on manually setting up your commit email as Gitpod misconfigures this by default, or it used to. Search for `gitpod github "@users.noreply.github.com"` to find out more about this.

Author: Looks like this can be explained with this link: https://www.gitpod.io/docs/configure/authentication/github

Reviewer 2: I'm in favor of adding the link to the gitpod docs instead, we should try not to end up writing a custom guide in these docs when we can link to the official docs for gitpod.

e.g. https://www.gitpod.io/docs/configure/authentication/github

Author: Looks like this can be addressed with a link to this page https://www.gitpod.io/docs/configure/authentication#how-to-use-a-private-github-email-or-custom-email-for-git-commits

Reviewer: There's no need for a guide, just need to mention that our custom GitPod config removes the commit email. Can then link to that page etc. No need to say how to do anything just that it needs to be done if you want to make a commit.

> For more information on Gitpod click [here](https://www.gitpod.io/).

Reviewer: Would be good if this linked to the docs directly and [link context was provided](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-refs.html).
